modules/mcuboot/CMakeLists.txt

```diff
@@ -524,27 +671,37 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
     else()
       # No network core update
       if(CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT)
+        if(CONFIG_XIP_SPLIT_IMAGE)
+          message(FATAL_ERROR "This build configuration is not supported:
+CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT cannot be used with CONFIG_XIP_SPLIT_IMAGE.")
+        endif()
+
         set(secondary_app_core_binary_name mcuboot_secondary_app_update.bin)
         set(generate_bin_files
-          ${PROJECT_BINARY_DIR}/${app_core_binary_name}
+          ${generate_bin_app_files}
           ${PROJECT_BINARY_DIR}/${secondary_app_core_binary_name}
-          )
+        )
         set(generate_script_params
           "${app_core_binary_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_APP_ADDRESS>"
           "${app_core_binary_name}version_MCUBOOT+XIP=${CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION}"
           "${app_core_binary_name}slot=0"
           "${secondary_app_core_binary_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_SECONDARY_APP_ADDRESS>"
           "${secondary_app_core_binary_name}version_MCUBOOT+XIP=${CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION}"
           "${secondary_app_core_binary_name}slot=1"
-          )
+        )
       else()
         set(generate_bin_files
-          ${PROJECT_BINARY_DIR}/${app_core_binary_name}
-          )
-        set(generate_script_params
-          "load_address=$<TARGET_PROPERTY:partition_manager,PM_APP_ADDRESS>"
-          "version_MCUBOOT=${CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION}"
+          ${generate_bin_app_files}
+        )
+
+        if(NOT CONFIG_XIP_SPLIT_IMAGE)
+          set(generate_script_params
+            "load_address=$<TARGET_PROPERTY:partition_manager,PM_APP_ADDRESS>"
+            "version_MCUBOOT=${CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSIO}" #CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION ?
```

Comment:
why not `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION`?  it is also repeated in above method like `generate_script_app_params`

Reply:
it's typo